res/values/arrays.xml

```diff
@@ -2,74 +2,74 @@
+    <item>grek_ell_gr_qwerty</item>
+    <item>hang_kor_kr_dubeolsik</item>
+    <item>hebr_heb_il_</item>
+    <item>hebr_heb_il_2</item>
+    <item>latn_ces_cz_qwertz</item>
+    <item>latn_deu_zz_bone</item>
+    <item>latn_deu_zz_neo2</item>
+    <item>latn_deu_de_qwertz</item>
+    <item>latn_deu_zz_qwertz</item>
+    <item>latn_eng_zz_colemak</item>
+    <item>latn_eng_zz_dvorak</item>
+    <item>latn_eng_us_qwerty</item>
+    <item>latn_spa_es_qwerty</item>
+    <item>latn_fra_fr_azerty</item>
+    <item>latn_lav_lv_qwerty</item>
+    <item>latn_hun_zz_qwerty</item>
+    <item>latn_hun_hu_qwertz</item>
+    <item>latn_nor_no_qwerty</item>
+    <item>latn_pol_pl_qwerty</item>
+    <item>latn_por_br_qwerty</item>
+    <item>latn_slk_sk_qwertz</item>
+    <item>latn_swe_se_qwerty</item>
+    <item>latn_vie_vi_qwerty</item>
+    <item>latn_tur_tr_qwerty</item>
     <item>custom</item>
   </string-array>
   <string-array name="pref_layout_entries">
     <item>None</item>
-    <item>AZERTY</item>
-    <item>QWERTY</item>
-    <item>QWERTY (Brasileiro)</item>
-    <item>QWERTY (Español)</item>
-    <item>QWERTY (Greek)</item>
-    <item>QWERTY (Korean)</item>
-    <item>QWERTY (Latvian)</item>
-    <item>QWERTY (Norwegian)</item>
-    <item>QWERTY (Swedish)</item>
-    <item>QWERTY (Hungarian)</item>
-    <item>QWERTY (Türkçe)</item>
-    <item>QWERTY (Polski)</item>
-    <item>QWERTY (Vietnamese)</item>
-    <item>ЙЦУКЕН (Русский)</item>
-    <item>QWERTZ</item>
-    <item>QWERTZ (Czech)</item>
-    <item>QWERTZ (Deutsch)</item>
-    <item>QWERTZ (Hungarian)</item>
-    <item>QWERTZ (Slovak)</item>
-    <item>Bulgarian (Phonetic Traditional)</item>
-    <item>Dvorak</item>
-    <item>Colemak</item>
-    <item>Neo 2</item>
-    <item>Bone</item>
-    <item>ЙЦУКЕН (Українська)</item>
-    <item>বাংলা</item>
-    <item>देवनागरी (हिंदी)-1</item>
-    <item>देवनागरी (हिंदी)-2</item>
-    <item>Hebrew SI-1452-1</item>
-    <item>Hebrew SI-1452-2</item>
-    <item>Arabic PC</item>
-    <item>Arabic ALT</item>
+    <item>الْعَرَبِيَّة: IBM [السَّعُودِيَّة]</item>
+    <item>الْعَرَبِيَّة: IBM (Mostafa Ayesh)</item>
+    <item>বাংলা: (Md Rasel Hossain)</item>
+    <item>бъ́лгарски: ЯВЕРТЪ</item>
+    <item>ру́сский: ЙЦУКЕН [Росси́я]</item>
+    <item>украї́нська: ЙЦУКЕН [Украї́на]</item>
+    <item>हिंदी: InScript [भारत]</item>
+    <item>हिंदी: (Luke Videckis)</item>
+    <item>ελληνικά: QWERTY [Ελλάδα]</item>
+    <item>한국어/조선말: 두벌식 [대한민국]</item>
+    <item>עִבְרִית: [יִשְׂרָאֵל]</item>
+    <item>עִבְרִית: [יִשְׂרָאֵל] 2</item>
+    <item>čeština: QWERTZ [Česko]</item>
+    <item>Deutsch: Bone</item>
+    <item>Deutsch: Neo 2</item>
+    <item>Deutsch: QWERTZ [Deutschland]</item>
+    <item>Deutsch: QWERTZ</item>
+    <item>English: Colemak</item>
+    <item>English: Dvorak</item>
+    <item>English: QWERTY [United States of America]</item>
+    <item>español: QWERTY [España]</item>
+    <item>français: AZERTY [France]</item>
```

Comment:
What about this: place the language name after the layou name?

Reply:
- I don't like the mention of the country and I'd prefer if it was removed unless there's variations of the same layout for the same language (for example, there's two QWERTZ for Deutsch). To avoid loosing the information, can you mention it in the layout `.xml` files ?

- I really want every items to be capitalized.

- Let's talk about grouping in a different PR. I'm not sure whether grouping make sense but it would be nice to try something.
  For now, sorted by the internal name is fine, which in your work is: script, country, lang, layout (with the programming layout at the top of course).